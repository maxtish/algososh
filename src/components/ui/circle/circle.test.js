import { Circle } from "./circle";
import renderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";

describe("Проверяем Circle при помощи снэпшотов корректную отрисовку элемента", () => {
  it("без буквы", () => {
    const cricleWithoutLetter = renderer.create(<Circle />);
    expect(cricleWithoutLetter).toMatchSnapshot();
  });

  it("с буквами", () => {
    const cricleWithLetter = renderer.create(<Circle letter="f" />);
    expect(cricleWithLetter).toMatchSnapshot();
  });

  it("с head", () => {
    const cricleWithHead = renderer.create(<Circle head={"top"} />);
    expect(cricleWithHead).toMatchSnapshot();
  });

  it("с react-элементом в head", () => {
    const cricleWithReactElInHead = renderer.create(
      <Circle head={<Circle letter="T" />} />
    );
    expect(cricleWithReactElInHead).toMatchSnapshot();
  });

  it("с tail", () => {
    const cricleWithTail = renderer.create(<Circle tail={"T"} />);
    expect(cricleWithTail).toMatchSnapshot();
  });

  it("с react-элементом в tail", () => {
    const cricleWithReactElInTail = renderer.create(
      <Circle tail={<Circle letter="T" />} />
    );
    expect(cricleWithReactElInTail).toMatchSnapshot();
  });

  it("с index", () => {
    const cricleWithIndex = renderer.create(<Circle index={1} />);
    expect(cricleWithIndex).toMatchSnapshot();
  });

  it("с пропом isSmall ===  true", () => {
    const cricleWithPropsIsSmall = renderer.create(<Circle isSmall={true} />);
    expect(cricleWithPropsIsSmall).toMatchSnapshot();
  });

  it("в состоянии default", () => {
    const cricleDefault = renderer.create(
      <Circle state={ElementStates.Default} />
    );
    expect(cricleDefault).toMatchSnapshot();
  });

  it("в состоянии changing", () => {
    const cricleChanging = renderer.create(
      <Circle state={ElementStates.Changing} />
    );
    expect(cricleChanging).toMatchSnapshot();
  });

  it("в состоянии modified", () => {
    const cricleModified = renderer.create(
      <Circle state={ElementStates.Modified} />
    );
    expect(cricleModified).toMatchSnapshot();
  });
});
