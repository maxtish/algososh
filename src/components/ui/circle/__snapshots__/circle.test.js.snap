// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверяем Circle при помощи снэпшотов корректную отрисовку элемента без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем Circle при помощи снэпшотов корректную отрисовку элемента в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем Circle при помощи снэпшотов корректную отрисовку элемента в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем Circle при помощи снэпшотов корректную отрисовку элемента в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем Circle при помощи снэпшотов корректную отрисовку элемента с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    top
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем Circle при помощи снэпшотов корректную отрисовку элемента с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Проверяем Circle при помощи снэпшотов корректную отрисовку элемента с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          T
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем Circle при помощи снэпшотов корректную отрисовку элемента с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          T
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Проверяем Circle при помощи снэпшотов корректную отрисовку элемента с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    T
  </div>
</div>
`;

exports[`Проверяем Circle при помощи снэпшотов корректную отрисовку элемента с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      f
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем Circle при помощи снэпшотов корректную отрисовку элемента с пропом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
